lib/Conversion/HandshakeToFIRRTL/HandshakeToFIRRTL.cpp

```diff
@@ -1585,12 +1591,15 @@ bool HandshakeBuilder::visitHandshake(MemoryOp op) {
     // Unpack store control.
     auto storeControlValid = storeControl[0];
 
-    // Create a subfield op to access this port in the memory.
     auto fieldName = storeIdentifier(i);
-    auto subfieldType = resultType.getElementType(fieldName).cast<FlipType>();
-    auto bundleType = subfieldType.getElementType().cast<BundleType>();
-    auto memBundle =
-        rewriter.create<SubfieldOp>(insertLoc, subfieldType, memOp, fieldName);
+    auto memBundle = memOp.getPortNamed(fieldName);
+    // If this is a write port, then it will be flipped. Strip that.
+    BundleType bundleType = memBundle.getType().dyn_cast<BundleType>();
+    if (!bundleType)
+      bundleType = memBundle.getType()
+                       .cast<FlipType>()
+                       .getElementType()
+                       .cast<BundleType>();
```

Comment:
Does this work?
```suggestion
    BundleType bundleType = memBundle.getType().getPassiveType().cast<BundleType>();
```

Reply:
Yeah, that should work. I'll switch it in a fixup. There may be an issue with the "data" field needing to be selectively flipped based on the kind, but that's not tripping a test.